test/sql/timestamp_functions_sql_test.cpp

```diff
@@ -86,5 +86,97 @@ TEST_F(TimestampFunctionsSQLTest, DateTruncTest) {
   txn_manager.CommitTransaction(txn);
 }
 
+TEST_F(TimestampFunctionsSQLTest, DatePartTest) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, txn);
+  catalog::Catalog::GetInstance()->Bootstrap();
+  txn_manager.CommitTransaction(txn);
+  // Create a t
+  txn = txn_manager.BeginTransaction();
+
+  TestingSQLUtil::ExecuteSQLQuery(
+      "CREATE TABLE foo(id integer, value timestamp);");
+  // Add in a testing timestamp
+  TestingSQLUtil::ExecuteSQLQuery(
+      "insert into foo values(3, '2016-12-07 13:26:02.123456-05');");
+
+  txn_manager.CommitTransaction(txn);
+
+  // Fetch values from the table
+  std::vector<ResultValue> result;
+  std::vector<FieldInfo> tuple_descriptor;
+  std::string error_message;
+  int rows_affected;
+
+  // wrong argument type
+  std::string test_query = "select date_part('123', value) from foo;";
+  TestingSQLUtil::ExecuteSQLQuery(test_query.c_str(), result, tuple_descriptor,
+                                  rows_affected, error_message);
+  EXPECT_EQ(result.size(), 0);
```

Comment:
You're not testing the return value here. In fact, for all tests here, can you use `TestingSQLUtil::ExecuteSQLQueryAndCheckResult` to verify the return value?

Reply:
The result of a few queries is empty and that is where the return value is not checked. Instead `EXPECT_EQ(result.size(), 0);` check is done. 

I think this was done in tests as
`TestingSQLUtil::ExecuteSQLQueryAndCheckResult` has a line `unsigned int rows = result.size() / tuple_descriptor.size();` which gives a floating point exception when the result is empty.

I can add a check whether the expected result is empty if `result.size() == 0`  in `TestingSQLUtil::ExecuteSQLQueryAndCheckResult`. Should I add it or is the current format fine?

I have changed other tests to use `TestingSQLUtil::ExecuteSQLQueryAndCheckResult`

Reply:
@pervazea What do you think?